apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,236 @@
+
+
+const REDIS_GUARD = new RedisGuard()
+const MAX_ANONYMOUS_REQUESTS_BY_DAY = Number(conf['MAX_ANONYMOUS_REQUESTS_FOR_IP_BY_DAY']) || MAX_ANONYMOUS_REQUESTS_FOR_IP_BY_DAY
+const IP_WHITE_LIST = conf.IP_WHITE_LIST ? JSON.parse(conf.IP_WHITE_LIST) : []
+
+const REQUIRED_USER_FIELDS_BY_TYPE = {
+    [RESIDENT]: ['phone'],
+    [STAFF]: ['phone', 'name', 'password'],
+}
+const SUPPORTED_USER_TYPES = [RESIDENT, STAFF]
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const ERRORS = {
+    REQUIRED_USER_DATA_IS_MISSING: (missingFields = []) => ({
+        mutation: 'signInUser',
+        variable: ['data', 'userData', missingFields[0]],
+        code: BAD_USER_INPUT,
+        type: 'REQUIRED_USER_DATA_IS_MISSING',
+        message: `Some required user data was missing: ${missingFields.join(', ')}`,
+        messageForUser: 'api.user.signInUser.REQUIRED_USER_DATA_IS_MISSING',
+    }),
+    NOT_SUPPORTED_USER_TYPE: {
+        mutation: 'signInUser',
+        variable: ['data', 'userType'],
+        code: BAD_USER_INPUT,
+        type: 'NOT_SUPPORTED_USER_TYPE',
+        message: `Only the following user types are supported: ${SUPPORTED_USER_TYPES.join(', ')}`,
+    },
+    DIFFERENT_PHONE_NUMBERS: {
+        mutation: 'signInUser',
+        variable: ['data', 'userData', 'phone'],
+        code: BAD_USER_INPUT,
+        type: 'DIFFERENT_PHONE_NUMBERS',
+        message: 'The verified phone number and the phone number from the payload cannot be different',
+    },
+    INVALID_PHONE_NUMBER: {
+        mutation: 'signInUser',
+        code: BAD_USER_INPUT,
+        type: 'INVALID_PHONE_NUMBER',
+        message: 'Invalid phone number',
+        messageForUser: 'api.user.signInUser.INVALID_PHONE_NUMBER',
+    },
+    USER_NOT_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'USER_NOT_FOUND',
+        message: 'User not found',
+        messageForUser: 'api.user.signInUser.USER_NOT_FOUND',
+    },
+    TOKEN_NOT_FOUND: {
+        mutation: 'signInUser',
+        variable: ['data', 'token'],
+        code: NOT_FOUND,
+        type: 'TOKEN_NOT_FOUND',
+        message: 'Token not found',
+    },
+    MULTIPLE_USERS_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'MULTIPLE_USERS_FOUND',
+        message: 'Multiple users found',
+    },
+    DV_VERSION_MISMATCH: {
+        query: 'signInUser',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'signInUser',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
```

Comment:
IMO those errors looks like something that might be reused in most of the domain models.